File: app/src/main/java/com/xeniac/warrantyroster_manager/LocaleModifier.java
package com.xeniac.warrantyroster_manager;

import android.content.Context;
import android.content.SharedPreferences;
import android.content.res.Configuration;
import android.content.res.Resources;
import android.util.DisplayMetrics;

import java.util.Locale;

public class LocaleModifier {

    public static void setLocale(Context context) {
        SharedPreferences localePrefs = context.getSharedPreferences(Constants.PREFERENCE_SETTINGS, Context.MODE_PRIVATE);
        String language = localePrefs.getString(Constants.PREFERENCE_LANGUAGE_KEY, "en");
        String country = localePrefs.getString(Constants.PREFERENCE_COUNTRY_KEY, "US");

        Resources resources = context.getResources();
        DisplayMetrics displayMetrics = resources.getDisplayMetrics();
        Configuration configuration = resources.getConfiguration();
        Locale newLocale = new Locale(language, country);
        Locale.setDefault(newLocale);
        configuration.setLocale(newLocale);
        resources.updateConfiguration(configuration, displayMetrics);
    }
}

File: app/src/main/java/com/xeniac/warrantyroster_manager/util/LocaleModifier.kt
package com.xeniac.warrantyroster_manager.util

import android.content.Context
import com.xeniac.warrantyroster_manager.util.Constants.Companion.PREFERENCE_COUNTRY_KEY
import com.xeniac.warrantyroster_manager.util.Constants.Companion.PREFERENCE_LANGUAGE_KEY
import com.xeniac.warrantyroster_manager.util.Constants.Companion.PREFERENCE_SETTINGS
import java.util.*

class LocaleModifier {
    companion object {
        fun setLocale(context: Context) {
            val localePrefs = context
                .getSharedPreferences(PREFERENCE_SETTINGS, Context.MODE_PRIVATE)
            val language = localePrefs.getString(PREFERENCE_LANGUAGE_KEY, "en").toString()
            val country = localePrefs.getString(PREFERENCE_COUNTRY_KEY, "US").toString()

            val resources = context.resources
            val displayMetrics = resources.displayMetrics
            val configuration = resources.configuration
            val newLocale = Locale(language, country)
            Locale.setDefault(newLocale)
            configuration.setLocale(newLocale)
            resources.updateConfiguration(configuration, displayMetrics)
        }
    }
}